File: Versionen/Datenstrukturen/BlueJ/04-Queue_ML/Queue.java

public class Queue<ContentType> {

    private QueueNode<ContentType> head = null;

    private QueueNode<ContentType> tail = null;

    public Queue() {
    }

    public void enqueue( ContentType pContentObject ) {
        QueueNode<ContentType> n = new QueueNode<ContentType>(pContentObject);
        if( isEmpty() ) {
            head = n;
            tail = n;
        } else {
            tail.setNext(n);
            tail = n;
        }
    }

    public void dequeue() {
        if( head != tail ) {
            head = head.getNext();
        } else if( !isEmpty() ) {
            head = null;
            tail = null;
        }
    }

    /**
     * Liefert das Inhaltsobjekt vom ersten Knotens der Schlange
     * vom Typ ContentType, falls die Schlange nicht leer ist,
     * ansonsten null.
     *
     * @return Das Inhaltsobjekt oder null.
     */
    public ContentType front() {
        if( !isEmpty() ) {
            return head.getContent();
        } else {
            return null;
        }
    }

    /**
     * Prüft, ob die Schlange leer ist.
     *
     * @return true, wenn die Schlange keine Knoten enthaelt
     */
    public boolean isEmpty() {
        return (head == null);
    }

}

File: Abiturklassen/Python/queue.py
# -*- coding: utf-8 -*-


class Queue(object):
    class QueueNode(object):
        def __init__(self, pContent):
            self._content = pContent
            self._nextNode = None

        def setNext(self, pNext):
            self._nextNode = pNext

        def getNext(self):
            return self._nextNode

        def getContent(self):
            return self._content

    def __init__(self):
        self._head = None
        self._tail = None

    def isEmpty(self):
        return self._head == None

    def enqueue(self, pContent):
        if pContent is not None:
            newNode = self.QueueNode(pContent)
            if self.isEmpty():
                self._head = newNode
                self._tail = newNode
            else:
                self._tail.setNext(newNode)
                self._tail = newNode

    def dequeue(self):
        if not self.isEmpty():
            self._head = self._head.getNext()
            if self.isEmpty():
                self._head = None
                self._tail = None

    def front(self):
        if self.isEmpty():
            return None
        else:
            return self._head.getContent()
